### PR TITLE
app: run every epoch

### PR DESCRIPTION
**Pull request by gsora**
Instead of hammering the beacon node once every second, run the lifecycle whenever an epoch ticks.

Also set the beacon node queries timeout to 10s instead of 1s.

category: refactor
ticket: none

### CODE REVIEW

File: app/bnapi/bnapi.go [hunk 1]
```diff
@@ -166,9 +167,19 @@ func MockBeaconNode(validators map[string]eth2v1.Validator) http.Handler {
 	}))
 
 	router.Handle("/eth/v1/config/deposit_contract", logHandler(func(writer http.ResponseWriter, request *http.Request) {
+		addressBytes, err := hex.DecodeString("07b39f4fde4a38bace212b546dac87c58dfe3fdc")
```
Comment (xenowits) [inline app/bnapi/bnapi.go:170]: nit: can declare the deposit address as a const
```
const depositContractAddr = "07b39f4fde4a38bace212b546dac87c58dfe3fdc"
```

Comment (dB2510) [inline app/bnapi/bnapi.go:170]: why are we using static values?

File: app/slot_ticker.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+	"github.com/obolnetwork/charon/app/eth2wrap"
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/core"
+)
+
+// newSlotTicker returns a blocking channel that will be populated with new slots in real time.
+// It is also populated with the current slot immediately.
+//
+// Taken from Charon's core/scheduler package.
```
Comment (dB2510) [inline app/slot_ticker.go:20]: can you add a permalink to that section in charon?

**Review by dB2510 — APPROVED** [review]

File: go.mod [hunk 3]
```diff
@@ -1,21 +1,24 @@
 module github.com/ObolNetwork/lido-dv-exit
 
-go 1.20
+go 1.21
+
+toolchain go1.21.4
```
Comment (xenowits) [inline go.mod:5]: why do we need a stricter toolchain version here?

**Review by xenowits — APPROVED** [review]

File: app/bnapi/bnapi.go (hunk 1 above)
Comment (gsora) [reply app/bnapi/bnapi.go:170]: this is a mock response, only needed to make go-eth2-client initialize correctly

File: app/slot_ticker.go (hunk 2 above)
Comment (gsora) [reply app/slot_ticker.go:20]: hmmm i can but that link might get stale fast if we end up refactoring scheduler, and result in 404 :/

File: go.mod (hunk 3 above)
Comment (gsora) [reply go.mod:5]: probably due to https://go.dev/doc/toolchain

it appeared after a `go mod tidy`

**Pull request merged**